projects/templates/projects/view_project.html

```diff
@@ -146,92 +146,174 @@ <h3>Project Members</h3>
+          <b>Email</b>: {{ u.email }}
+        </div>
+        <div class="col-xs-4 col-sm-4">
+          <div class="row" align="center">
+            <div
+              class="col-xs-{% if can_manage_project_membership %}2{% else %}6{% endif %} col-sm-{% if can_manage_project_membership %}2{% else %}6{% endif %} col-lg-{% if can_manage_project_membership %}2{% else %}6{% endif %}"
+              align="right">
+              <b>Role: </b>
+            </div>
+            <div
+              class="col-xs-{% if can_manage_project_membership %}3{% else %}6{% endif %} col-sm-{% if can_manage_project_membership %}3{% else %}6{% endif %} col-lg-{% if can_manage_project_membership %}3{% else %}6{% endif %}"
+              align="left">
+              {% if can_manage_project_membership %}
+                <form role="form" method="post" style="display:inline">
+                  {% csrf_token %}
+                  <input type="hidden" name="user_ref" value="{{ u.username }}">
+                  <select id="role_{{ u.id }}" name="user_role" class="form-select role_selector" data-initial="{{ u.role }}">
+                    {% for role in roles %}
+                      <option value="{{ role }}" {% if role == u.role %}selected disabled{% endif %}>{{ role }}</option>
+                    {% endfor %}
+                  </select>
+                </form>
+              {% else %}
+                {{u.role}}
+              {% endif %}
+            </div>
+            {% if can_manage_project_membership %}
+              <div class="col-xs-2 col-sm-2 col-lg-2">
+                <button type="submit" id="submit_role_{{ u.id }}" class="btn btn-xs btn-primary" name="change_role">Submit</button>
+              </div>
+            {% endif %}
+            {% if can_manage_project_membership %}
+              <div class="col-xs-2 col-sm-2 col-lg-2">
+                    <button type="button" onclick="resetRole('role_{{ u.id }}')" id="cancel_role_{{ u.id }}" class="btn btn-xs btn-danger" >Cancel</button>
+              </div>
+            {% endif %}
+          </div>
+        </div>
+        {% if can_manage_project_membership %}
+          <form role="form" method="post">
             {% csrf_token %}
-            <input type="hidden" name="join_request" value="{{ join_request.id }}">
-            <button type="submit" class="btn btn-xs btn-success btn-block" name="accept_join_request">Accept</button>
-            <button type="submit" class="btn btn-xs btn-danger btn-block" name="reject_join_request">Reject</button>
+            <div class="col-xs-4 col-sm-4 col-lg-4">
+              <div class="row">
+                <div class="col-xs-5 col-sm-5 col-lg-5">
+                  <b>Usage</b>: <p style="display: inline-block">{{u.su_used}}/</p>
+                  <p class="su-budget-display" style="display: inline-block">{{ u.su_budget }}</p>
+                </div>
+                <div class="col-xs-5 col-sm-5 col-lg-5">
+                  <input
```

Comment:
How usable is this? It seems like it would be tricky to precisely set the slider. Also steps of 1000 is a lot, since our allocations only have 20000 SUs, and a class might have more than 20 people.

Reply:
Maybe 500 would be sufficient step.